src/arp.rs

```diff
@@ -1,59 +1,65 @@
 //! Handles parsing of Arp pakets
 
-use nom::Endianness::Big;
-use nom::{be_u8, IResult};
-
+use nom::number;
+use nom::IResult;
 use std::net::Ipv4Addr;
 
-use crate::ethernet::{to_mac_address, MacAddress};
-use crate::ipv4::to_ipv4_address;
+use crate::ethernet;
+use crate::ethernet::MacAddress;
+use crate::ipv4;
 
 #[derive(Debug, PartialEq, Eq)]
-#[cfg_attr(feature = "derive", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "derive", derive(serde::Serialize, serde::Deserialize))]
 pub enum HardwareAddressType {
     Ethernet,
     Other(u16),
 }
 
-fn to_hw_addr_type(i: u16) -> Option<HardwareAddressType> {
```

Comment:
I removed the Option, I don't see the point here. I also implement `From` trait because this conversion can't fail.